modules/4337/contracts/Safe4337Module.sol

```diff
@@ -258,16 +258,17 @@ contract Safe4337Module is IAccount, HandlerContext, CompatibilityFallbackHandle
     function _validateSignatures(PackedUserOperation calldata userOp) internal view returns (uint256 validationData) {
         (bytes memory operationData, uint48 validAfter, uint48 validUntil, bytes calldata signatures) = _getSafeOp(userOp);
 
+        // checkSignatures function in Safe contract does not force a fixed size on signature length. A malicious actor can append additional bytes in the signature than needed and force gas usage to reach verificationGasLimit. _checkSignatureLength ensures that there are no additional bytes in the signatures than required.
+        if (!_checkSignatureLength(signatures, ISafe(payable(userOp.sender)).getThreshold()))
+            return _packValidationData(true, validUntil, validAfter);
+
         bool validSignature;
         try ISafe(payable(userOp.sender)).checkSignatures(keccak256(operationData), operationData, signatures) {
             validSignature = true;
```

Comment:
My issue with this is the same as before, we now have multiple exit branches which feels a bit icky IMO. This also has signature verification gotcha's in that if you pass an invalid length signature for gas estimations, it will not estimate correctly (so same problem you were having, but other way around.

Maybe we can do:

```
        validSignature = _checkSignatureLength(signatures, ISafe(payable(userOp.sender)).getThreshold()));
        try ISafe(payable(userOp.sender)).checkSignatures(keccak256(operationData), operationData, signatures) {
        } catch {
            validSignature = false;
        }
```

Reply:
> This also has signature verification gotcha's in that if you pass an invalid length signature for gas estimations, it will not estimate correctly (so same problem you were having, but other way around.

I think this is fine. It still applies the other way around. If you pass an invalid length signature (but shorter), it will not estimate correctly because of the checks in the safe contract or the verifier if you use passkeys

Reply:
Right - but it is fairly trivial to always call `checkSignatures` (which is what I'm suggesting).